File: 974-reorder-data-in-log-files/reorder-data-in-log-files.js
/**
 * @param {string[]} logs
 * @return {string[]}
 */
var reorderLogFiles = function(logs) {
    const digits = [],
          letters = []
    
    for (const log of logs) {
        if (isFinite(log.split(' ')[1])) digits.push(log)  // digit log
        else letters.push(log)  // letter log
    }
    
	// Sorting the letters log based on condition of lexio order without and with key (example: 'let1')
    letters.sort((a,b) => {
        let l1 = a.split(' ').slice(1).join(' '),
            l2 = b.split(' ').slice(1).join(' ')
        
        if (l1 === l2) return a > b ? 1 : -1
        return l1 > l2 ? 1 : -1
    })
    
    return [...letters, ...digits]
};

File: 974-reorder-data-in-log-files/reorder-data-in-log-files.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        letter_logs = []
        digit_logs = []

        for log in logs:
            if log.split()[1].isdigit():
                digit_logs.append(log)
            else:
                letter_logs.append(log)

        letter_logs.sort(key=lambda log: (log.split(' ', 1)[1], log.split(' ', 1)[0]))

        return letter_logs + digit_logs